src/Engines/Native/nativeShaderProcessors.ts

```diff
@@ -15,6 +15,9 @@ export class NativeShaderProcessor extends WebGL2ShaderProcessor {
         if (!isFragment && !(<NativeEngine>engine).homogeneousDepth) {
             const lastClosingCurly = code.lastIndexOf("}");
             code = code.substring(0, lastClosingCurly);
+            if (!(<NativeEngine>engine).originBottomLeft) {
+                code += "gl_Position.y *= -1.;\n";
+            }

```

Comment:
I thought we moved all of the shader manipulation logic to the native side? When did we add this back? :)

If we moved to the native side, then you don't need to expose `homogeneousDepth` and `originBottomLeft` anymore, is that right?

Reply:
How hard would it be to do it in shader traversals @Drigax ? I didn't look at glslang yet

Reply:
I'm not sure you need shader traversal. We can manipulate the strings just the same as you are doing here.

Reply:
This should not be hard to do in a traverser. All it should require is figuring out how to express this operation (most easily done by inspecting an example in the debugger) then creating a helper to replicate that, effectively injecting this line into the AST when appropriate.